app/views/contexts/preview/preview_question.py

```diff
@@ -0,0 +1,66 @@
+from typing import Any, Union
+
+from werkzeug.datastructures import ImmutableDict
+
+from app.questionnaire import QuestionnaireSchema
+
+
+class PreviewQuestion:
+    def __init__(
+        self,
+        *,
+        block: ImmutableDict,
+    ):
+        self._block = block
+        self._block_id = block.get("id")
+        self._question = self.resolve_variants().get("question")
+        # resolve_variants returns same type as placeholder_renderer.render which is dict[str, Any] hence all type ignores below
+        self._title = self._question.get("title")  # type: ignore
+        self._answers = self._build_answers()
+        self._descriptions = self._question.get("description")  # type: ignore
+        self._guidance = self._question.get("guidance")  # type: ignore
+        self._instruction = self._question.get("instruction")  # type: ignore
+
+    def _build_answers(self) -> list[dict]:
+        answers_list = []
+        if answers := self._question.get("answers"):  # type: ignore
+            for answer in answers:
+                if options := answer.get("options"):
+                    options_list = [option["label"] for option in options]
+                    answer_dict = {"options": options_list}
+                else:
+                    answer_dict = {"label": answer["label"]}
+                if description := answer.get("description"):
+                    answer_dict["description"] = description
+                if instruction := answer.get("instruction"):
+                    answer_dict["instruction"] = instruction
+                if guidance := answer.get("guidance"):
+                    answer_dict["guidance"] = guidance
+                if answer.get("type") == "TextArea" and (
+                    length := answer.get("max_length")
+                ):
+                    answer_dict["max_length"] = length
+
+                answers_list.append(answer_dict)
+        return answers_list
+
+    def serialize(self) -> dict[str, Union[str, dict, Any]]:
+        return {
+            "id": self._block_id,
+            "title": self._title,
+            "answers": self._answers,
+            "descriptions": self._descriptions,
+            "guidance": self._guidance,
+            "instruction": self._instruction,
+        }
+
+    def resolve_variants(self) -> dict[str, Any]:
+        output_block: dict = QuestionnaireSchema.get_mutable_deepcopy(self._block)
```

Comment:
Otherwise it's immutable dict and we cannot pop and push elements of the block, I just follow what we do in other variants resolution methods and simplify for our needs.

Reply:
The other instance where we use `get_mutable_deepcopy()` in this method is for cosmetic reasons, part of the return context had some elements that were immutable dicts.

Reply:
Ok, first instance of `get_mutable_deepcopy()` now removed.